common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/OneForOneBlockPusher.java

```diff
@@ -77,42 +81,55 @@ public OneForOneBlockPusher(
 
     @Override
     public void onSuccess(ByteBuffer response) {
-      // On receipt of a successful block push
-      listener.onBlockPushSuccess(blockId, new NioManagedBuffer(ByteBuffer.allocate(0)));
+      BlockPushReturnCode returnCode =
+        (BlockPushReturnCode) BlockTransferMessage.Decoder.fromByteBuffer(response);
+      // If the return code is not SUCCESS, the server has responded some error code. Handle
+      // the error accordingly.
+      if (returnCode.returnCode != ReturnCode.SUCCESS.id()) {
+        checkAndFailRemainingBlocks(index, new BlockPushNonFatalFailure(
+          BlockPushNonFatalFailure.getReturnCode(returnCode.returnCode)));
```

Comment:
Move this as a helper static method into `BlockPushNonFatalFailure` ? It is a repeating pattern to create `BlockPushNonFatalFailure` given a `ReturnCode` id.